lib/DelegatedAdminsMiddleware.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace OCA\GroupFolders;
+
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Http\Response;
+use OCP\AppFramework\Http\TemplateResponse;
+use OCP\AppFramework\Middleware;
+use OCP\IAppConfig;
+use OCP\IGroup;
+use OCP\IGroupManager;
+use OCP\IRequest;
+use OCP\IUserSession;
+use Psr\Log\LoggerInterface;
+
+class DelegatedAdminsMiddleware extends Middleware {
+
+    /** @var string */
+    private $appName;
+
+    /** @var IAppConfig */
+    private $appConfig;
+
+    /** @var IGroupManager */
+    private $groupManager;
+
+    /** @var IRequest */
+    private $request;
+
+    /** @var IUserSession */
+    private  $userSession;
+
+    /**
+     *
+     * @param string $appName
+     * @param IAppConfig $appConfig
+     * @param IgroupManager $groupManager
+     * @param IUserSession $userSession
+     *
+     */
+    public function __construct(string $appName,
+        IAppConfig $appConfig,
+        IgroupManager $groupManager,
+        IRequest $request,
+        IUserSession $userSession
+        ) {
+            $this->appName = $appName;
+            $this->appConfig = $appConfig;
+            $this->groupManager = $groupManager;
+            $this->request = $request;
+            $this->userSession = $userSession;
+    }
+
+    /**
+     *
+     * {@inheritDoc}
+     * @see \OCP\AppFramework\Middleware::beforeController()
+     *
+     * Throws an error when the user is not allowed to use the app's APIs
+     *
+     */
+    public function beforeController($controller, $methodName) {
+
+        // method 'aclMappingSearch' implements its own access control and method 'isAdmin' must be accesible by everyone
+        if ($methodName !== 'aclMappingSearch' && $methodName !== 'isAdmin') {
+            // Get allowed groups from app's config
+            $appConfigKeys = $this->appConfig->getValues($this->appName, false);
```

Comment:
No need to fetch all, you can just sue https://github.com/nextcloud/server/blob/2a054e6c04e0a40421510eb889cbf59f153c5177/lib/public/AppFramework/Services/IAppConfig.php#L61

Reply:
done